### PR TITLE
Uploading step logs to Results as well 

### PR DESCRIPTION
**Pull request by yacaovsnc**
Also create step logs in Results.  Best effort.

### CODE REVIEW

File: src/Runner.Common/JobServer.cs [hunk 1]
```diff
@@ -30,7 +30,8 @@ public interface IJobServer : IRunnerService, IAsyncDisposable
         Task<TaskLog> AppendLogContentAsync(Guid scopeIdentifier, string hubName, Guid planId, int logId, Stream uploadStream, CancellationToken cancellationToken);
         Task AppendTimelineRecordFeedAsync(Guid scopeIdentifier, string hubName, Guid planId, Guid timelineId, Guid timelineRecordId, Guid stepId, IList<string> lines, long? startLine, CancellationToken cancellationToken);
         Task<TaskAttachment> CreateAttachmentAsync(Guid scopeIdentifier, string hubName, Guid planId, Guid timelineId, Guid timelineRecordId, String type, String name, Stream uploadStream, CancellationToken cancellationToken);
-        Task CreateStepSymmaryAsync(string planId, string jobId, string stepId, string file, CancellationToken cancellationToken);
+        Task CreateStepSymmaryAsync(string planId, string jobId, Guid stepId, string file, CancellationToken cancellationToken);
```
Comment (konradpabjan) [inline src/Runner.Common/JobServer.cs:33]:
```suggestion
        Task CreateStepSummaryAsync(string planId, string jobId, Guid stepId, string file, CancellationToken cancellationToken);
```

Typo 🤔 You'll have to change the name in a few places

File: src/Runner.Common/JobServerQueue.cs [hunk 2]
```diff
@@ -481,16 +502,13 @@ private async Task ProcessSummaryUploadQueueAsync(bool runOnce = false)
...
                             Trace.Error(ex);
-                        }
-                        finally
-                        {
                             errorCount++;
                         }
                     }
 
-                    Trace.Info("Tried to upload {0} summary files to results, success rate: {1}/{0}.", filesToUpload.Count, filesToUpload.Count - errorCount);
+                    Trace.Info("Tried to upload {0} files to results, success rate: {1}/{0}.", filesToUpload.Count, filesToUpload.Count - errorCount);
```
Comment (konradpabjan) [inline src/Runner.Common/JobServerQueue.cs:511]:
```suggestion
                    Trace.Info("Tried to upload {0} file(s) to results, success rate: {1}/{0}.", filesToUpload.Count, filesToUpload.Count - errorCount);
```

Same as one of the earlier Nits

File: src/Sdk/WebApi/WebApi/ResultsHttpClient.cs [hunk 3]
```diff
@@ -85,6 +112,40 @@ private async Task StepSummaryUploadCompleteAsync(string planId, string jobId, s
...
+            var request = new StepLogsMetadataCreate()
+            {
+                WorkflowJobRunBackendId= jobId,
+                WorkflowRunBackendId= planId,
+                StepBackendId = stepId.ToString(),
+                UploadedAt = timestamp,
+                LineCount = lineCount,
+            };
+
+            var stepLogsUploadCompleteRequest = new Uri(m_resultsServiceUrl, "twirp/results.services.receiver.Receiver/CreateStepLogsMetadata");
```
Comment (konradpabjan) [inline src/Sdk/WebApi/WebApi/ResultsHttpClient.cs:127]: Thoughts on putting `"twirp/results.services.receiver.Receiver/CreateStepLogsMetadata"`, `"twirp/results.services.receiver.Receiver/GetStepLogsSignedBlobURL"` and `"twirp/results.services.receiver.Receiver/CreateStepSummaryMetadata"` into some list of constants at the bottom of the file? Maybe even a separate file.

I imagine in the future we'll have more of these types of URLs so a single place for all of them now might make sense 🤔

File: src/Runner.Common/JobServerQueue.cs [hunk 4]
```diff
@@ -804,6 +822,34 @@ private async Task UploadSummaryFile(SummaryUploadFileInfo file)
...
+            {
+                if (uploadSucceed && file.DeleteSource)
+                {
+                    try
+                    {
+                        File.Delete(file.Path);
+                    }
+                    catch (Exception ex)
+                    {
+                        Trace.Info("Catch exception during delete success results uploaded step log file.");
```
Comment (konradpabjan) [inline src/Runner.Common/JobServerQueue.cs:847]:
```suggestion
                        Trace.Info("Exception encountered during deletion of already successfully uploaded step log file to results.");
```

Was having a had time understanding this. We're effectively deleting a temporary file from my understanding

File: src/Runner.Common/JobServerQueue.cs [hunk 5]
```diff
@@ -804,6 +822,34 @@ private async Task UploadSummaryFile(SummaryUploadFileInfo file)
                 }
             }
         }
+
+        private async Task UploadResultsStepLogFile(ResultsUploadFileInfo file)
+        {
+            bool uploadSucceed = false;
+            try
+            {
+                Trace.Info($"Starting to step log file to results service {file.Name}, {file.Path}");
```
Comment (konradpabjan) [inline src/Runner.Common/JobServerQueue.cs:831]:
```suggestion
                Trace.Info($"Starting upload of step log file to results service {file.Name}, {file.Path}");
```

**Review by konradpabjan — DISMISSED** [review]
Looks good to me! 🚀 Just a bunch of small things. Mostly grammar tweaks to make it easier to read any logs/messages

File: src/Runner.Common/JobServerQueue.cs [hunk 6]
```diff
@@ -459,19 +469,30 @@ private async Task ProcessSummaryUploadQueueAsync(bool runOnce = false)
...
                         try
                         {
-                            await UploadSummaryFile(file);
+                            if (String.Equals(file.Type, ChecksAttachmentType.StepSummary, StringComparison.OrdinalIgnoreCase))
+                            {
+                                await UploadSummaryFile(file);
+                            }
+                            else if (String.Equals(file.Type, CoreAttachmentType.ResultsLog, StringComparison.OrdinalIgnoreCase))
+                            {
+                                if (file.RecordId != _jobTimelineRecordId)
```
[the remark this answers is not in the document]
Comment (yacaovsnc) [reply src/Runner.Common/JobServerQueue.cs:486]: We will relax this in next epic.

[the remark this answers is not in the document]
Comment (konradpabjan) [reply src/Runner.Common/JobServerQueue.cs:486]: Thanks! I would potentially just add a comment that for now we're only uploading step logs and job logs will be addressed in the future

File: src/Sdk/WebApi/WebApi/ResultsHttpClient.cs (hunk 3 above)
Comment (yacaovsnc) [reply src/Sdk/WebApi/WebApi/ResultsHttpClient.cs:127]: Let me sit on this for a bit.  I do want to encapsulate all the logic in a single method, so I don't have to jump around when reading code.  We should pull this out to a common place if it's used in multiple places.  Let me see if any other places would benefit from referencing those consts.

**Review by thboop — APPROVED** [review]
LGTM

File: src/Sdk/WebApi/WebApi/ResultsHttpClient.cs [hunk 7]
```diff
@@ -85,6 +112,40 @@ private async Task StepSummaryUploadCompleteAsync(string planId, string jobId, s
             }
         }
 
+        private async Task StepLogUploadCompleteAsync(string planId, string jobId, Guid stepId, long lineCount, CancellationToken cancellationToken)
+        {
+            var timestamp = DateTime.UtcNow.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK");
```
[the remark this answers is not in the document]
Comment (yacaovsnc) [reply src/Sdk/WebApi/WebApi/ResultsHttpClient.cs:117]: Cool!  I will need to make a runner PR again very soon for uploading job logs. 😅 

I will move those hardcoded values to the const file then.

**Pull request merged**